test-avocado/README.md

```diff
@@ -0,0 +1,79 @@
+# Porting tests to Avocado
+
+The test suite in this directory is going to replace the old one as
+soon as we can manage.
+
+The rules for tests are a bit different:
+
+ * One or more machines are dedicated to running all our tests: The
+   machines are initialized specifically for this purpose, and will
+   not be reused for anything else after all tests have been run.
+
+ * A test runs as root directly in one of the machines dedicated to
+   the test run.
+
+ * A test is allowed to drastically modify the machine, as long as
+   that doesn't break other tests.
+
+ * The tests should aim to be independent from each other: Although a
+   test run is deterministic and the tests are always run in the same
+   order, implicit or explicit dependencies between tests should be
+   avoided.  If they can't easily be avoided, they need to be
+   documented, of course.
+
+ * This means that tests should clean up after themselves, and restore
+   the machine into the same configuration as it was before the test.
+   (Of course, a test should not aim for rootkit-level stealthiness:
+   It can and should leave entries in the journal behind, etc.)
+
+More concretely:
+
+ * When a test starts, the cockpit socket is already active, but
+   cockpit itself is not running.  After a test, cockpit will be
+   stopped.
+
+ * A test is a Python file in .../test-avocado modeled after
+   checkexample-foo.py.
+
+ * Previously, we have combined multiple tests into one file, like
+   check-storage, which has 12 tests in it.  Now we need to split them
+   out into 12 individual Python files, maybe with a common support
+   module.
+
+ * Each test is a class derived from cockpit.Test, which in turn is
+   derived from avocado.test.Test.
+
+ * The test class normally shouldn't define the usual 'setup',
+   'action', and 'cleanup' methods.
+
+   Instead, it should define a 'test' method, and use
+   cockpit.Test.atcleanup to register any cleanup actions it wants.
+   The cockpit.Test class has some convenience functions such as
+   'replace_file' that should make this easier.  We should define more
+   convenience methods while porting.
+
+ * If you do want to define 'setup' or 'cleanup', make sure to call
+   cockpit.Test.setup and cockpit.Test.cleanup as appropriate as part
+   of it.
+
+ * Use 'self.browser' to get an instance of the old Browser class.  It
+   works exactly as before.
```

Comment:
I think it might be good to write a separate readme for transitioning from the old tests to the new.

Why should a current readme document a version that's no longer used?